src/MoonLight/Nodes/Drivers/parlio.cpp

```
@@ -384,21 +400,22 @@ uint8_t IRAM_ATTR __attribute__((hot)) show_parlio(uint8_t* parallelPins, uint32
-  create_transposed_led_output_optimized(parallel_buffer_remapped, parallel_buffer_repacked, leds_per_output, outputs, isRGBW, offSetR, offsetG, offsetB);
+  create_transposed_led_output_optimized(parallel_buffer_remapped, parallel_buffer_repacked, leds_per_output, outputs, components, offSetR, offsetG, offsetB, offsetW);
 
   // Calculate the exact size of ONE PIXEL's data in bits and bytes.
-  const uint32_t symbols_per_pixel = isRGBW ? 128 : 96;
+  const uint32_t symbols_per_pixel = components * 32;  // isRGBW ? 128 : 96;
   const uint32_t bits_per_pixel = symbols_per_pixel * parlio_config.data_width;
   const uint32_t bytes_per_pixel = (bits_per_pixel + 7) / 8;
   const uint32_t HW_MAX_BYTES_PER_CHUNK = parlio_config.max_transfer_size;
   const uint16_t max_leds_per_chunk = (bytes_per_pixel > 0) ? (HW_MAX_BYTES_PER_CHUNK / bytes_per_pixel) : 0;
-  const uint8_t num_chunks = (leds_per_output + max_leds_per_chunk - 1) / max_leds_per_chunk;
+  const uint8_t num_chunks = (max_leds_per_output + max_leds_per_chunk - 1) / max_leds_per_chunk;
```

Cppcheck (2.19.0)

[error] Division by zero.

(zerodiv)